### ISSUE TITLE
[Domino Visual Editor] Maven Build Fails to Start

### ISSUE THREAD

**Issue by JoelProminic** [prominic/Moonshine-IDE#834]
I noticed this error with the macOS Non-Sandbox Moonshine 3.1.0 build that was released today:
1. Create a new Visual Editor Project, and choose Domino NSF Database Visual Editor.  I also notice the same bug when building an existing project.
2. Open nsfs/nsf-moonshine/odp/Forms/%project_name%.form.  This should probably open automatically or otherwise be made easier to find.
3. Add some controls to the form and save.  Make sure to use unique IDs for each field.
4. Click Project > Build with Apache Maven
5. BUG:  Moonshine shows this error on the console:

    : Error #1009
    : TypeError: Error #1009
    : 	at actionScripts.plugins.maven::MavenBuildPlugin/start()
    : 	at actionScripts.plugins.maven::MavenBuildPlugin/prepareStart()
    : 	at actionScripts.plugins.maven::MavenBuildPlugin/startConsoleBuildHandler()
    : 	at flash.events::EventDispatcher/dispatchEvent()
    : 	at actionScripts.utils::KeyboardShortcutManager/dispatch()
    : 	at actionScripts.utils::KeyboardShortcutManager/stopEvent()
    : 	at actionScripts.ui.menu::MenuPlugin/redispatch()
    :
    : Click here to Report a Bug

@piotrzarzycki21, could this be related to some of your recent work on #833 and #806?

This breaks the Domino Visual Editor project in the release (or at least requires manual builds).   If this is an easy fix, we could consider making a quick update to the 3.1.0 release.  Otherwise, we can include this in 3.2.0 (or a 3.1.1 release), but we should get this released quickly.

**Comment by JoelProminic** [comment]
It looks like this is part of a more general Maven problem:  #837

**Comment by piotrzarzycki21** [comment]
Here is stacktrace from build after fix provided as part of #837. Above issue seems to be resolved, but I'm getting different one. I believe this can be closed:

```
: Starting Maven build...
: Maven path: /Applications/apache-maven-3.6.0
: Maven build directory: /Users/piotr/Downloads/Tests/DominoNsfNewVisualEditorProject
: Command: /Applications/apache-maven-3.6.0/bin/mvn install
: [INFO
: ] Scanning for projects...
: [W
: ARNING] 
: [WARNING] Some problems were encountered while building the effective model for org.openntf.nsfodp:nsf-moonshine-domino:domino-nsf:1.0.0
: [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-jar-plugin is miss
: ing. @ org.openntf.nsfodp:nsf-moonshine:1.0.0, /Users/piotr/Downloads/Tests/DominoNsfNewVisualEditor
: Project/pom.xml, line 210, column 15
: [WARNING] 
: [WARNING] Some problems were encountered while building the effective model for org.openntf.nsfodp:nsf-moonshine:pom:1.0.0
: [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-jar-plugin is missing. @ line 210, column 15
: [WARNING] 
: [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
: [WARNING] 
: [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
: [WARNING] 
: [WARNING] The project org.openntf.nsfodp:nsf-moonshine:pom:1.0.0 uses prerequisites which is only intended for maven-plugin projects but not for non maven-plugin projects. For such purposes you should use the maven-enforcer-plugin. See https://maven.apache.org/enforcer/enforcer-rules/requireMavenVersion.html
: [INFO] ------------------------------------------------------------------------
: [INFO] Reactor Build Order:
: [INFO] 
: [INFO] nsf-moonshine                                                      [pom]
: [INFO] nsf-moonshine-domino                                        [domino-nsf]
: [INFO] 
: [INFO] ------------------< org.openntf.nsfodp:nsf-moonshine >------------------
: [INFO] Building nsf-moonshine 1.0.0                                       [1/2]
: [INFO] --------------------------------[ pom ]---------------------------------
: [INFO] 
: [INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven-and-java) @ nsf-moonshine ---
: [
: INFO] 
: [INFO] --- tycho-source-plugin:1.2.0:plugin-source (plugin-source)
:  @ nsf-moonshine ---
: [INFO] 
: [
: INFO] --- maven-install-plugin:2.5.2:install (default-install) @ nsf-moonshine ---
: [INFO] Installing /Users/piotr/Downloads/Tests/DominoNsfNewVisualEditorProject/pom.xml to /Users/piotr/.m2/repository/org/openntf/nsfodp/nsf-moonshine/1.0.0/nsf-moonshine-1.0.0.pom
: [INFO] 
: [INFO] --------------< org.openntf.nsfodp:nsf-moonshine-domino >---------------
: [INFO] Building nsf-moonshine-domino 1.0.0                                [2/2]
: [INFO] -----------------------------[ domino-nsf ]-----------------------------
: [INFO] 
: [INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven-and-java) @ nsf-moonshine-domino ---
: [INFO] 
: [INFO] --- nsfodp-maven-plugin:3.4.0:generate-pde-structure (default-generate-pde-structure) @ nsf-moonshine-domino ---
: [INFO] 
: [INFO] --- nsfodp-maven-plugin:3.4.0:create-source-folders (default-create-source-folders) @ nsf-moonshine-domino ---
: [INFO] Generating missing source folder odp/Code/Java
: [INFO] 
: [INFO] --- nsfodp-maven-plugin:3.4.0:generate-metadata (default-generate-metadata) @ nsf-moonshine-domino ---
: [INFO] 
: [INFO] --- tycho-source-plugin:1.2.0:plugin-source (plugin-source) @ nsf-moonshine-domino ---
: [INFO] 
: [INFO] --- nsfodp-maven-plugin:3.4.0:compile (default-compile) @ nsf-moonshine-domino ---
: [INFO] ------------------------------------------------------------------------
: [INFO] Reactor Summary for nsf-moonshine 1.0.0:
: [INFO] 
: [INFO] nsf-moonshine ...................................... SUCCESS [  1.104 s]
: [INFO] nsf-moonshine-domino ............................... FAILURE [  0.407 s]
: [INFO] ------------------------------------------------------------------------
: [INFO] BUILD FAILURE
: [INFO] ------------------------------------------------------------------------
: [INFO] Total time:  4.763 s
: [INFO] Finished at: 2021-06-08T13:39:11+02:00
: [INFO] ------------------------------------------------------------------------
: [ERROR] Failed to execute goal org.openntf.maven:nsfodp-maven-plugin:3.4.0:compile (default-compile) on project nsf-moonshine-domino: Execution default-compile of goal org.openntf.maven:nsfodp-maven-plugin:3.4.0:compile failed: Either the notesProgram or compilerServerUrl property must be supplied -> [Help 1]
: [ERROR] 
: [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
: [ERROR] Re-run Maven using the -X switch to enable full debug logging.
: [ERROR] 
: [ERROR] For more information about the errors and possible solutions, please read the following articles:
: [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
: [ERROR] 
: [ERROR] After correcting the problems, you can resume the build with the command
: [ERROR]   mvn <goals> -rf :nsf-moonshine-domino
: Maven build has been terminated with exit code: 1
```

**Comment by JoelProminic** [comment]
This seems to be fixed.  The build ran without errors for me.  Unfortunately, there are some manual steps required to get this to build, so the remaining errors are not surprising..  It looks like you missed at least the global settings.xml update (we are trying to remove this with #835).
